pythonforandroid/bootstraps/common/build/templates/Service.tmpl.java

```diff
@@ -20,50 +13,25 @@ public int startType() {
     }
     {% endif %}
 
-    {% if not foreground %}
-    @Override
-    public boolean canDisplayNotification() {
-        return false;
-    }
-    {% endif %}
-
-    @Override
-    protected void doStartForeground(Bundle extras) {
-        Notification notification;
-        Context context = getApplicationContext();
-        Intent contextIntent = new Intent(context, PythonActivity.class);
-        PendingIntent pIntent = PendingIntent.getActivity(context, 0, contextIntent,
-            PendingIntent.FLAG_UPDATE_CURRENT);
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB) {
-            notification = new Notification(
-                context.getApplicationInfo().icon, "{{ args.name }}", System.currentTimeMillis());
-            try {
-                // prevent using NotificationCompat, this saves 100kb on apk
-                Method func = notification.getClass().getMethod(
-                    "setLatestEventInfo", Context.class, CharSequence.class,
-                    CharSequence.class, PendingIntent.class);
-                func.invoke(notification, context, "{{ args.name }}", "{{ name| capitalize }}", pIntent);
-            } catch (NoSuchMethodException | IllegalAccessException |
-                     IllegalArgumentException | InvocationTargetException e) {
-            }
-        } else {
-            Notification.Builder builder = new Notification.Builder(context);
-            builder.setContentTitle("{{ args.name }}");
-            builder.setContentText("{{ name| capitalize }}");
-            builder.setContentIntent(pIntent);
-            builder.setSmallIcon(context.getApplicationInfo().icon);
-            notification = builder.build();
-        }
-        startForeground({{ service_id }}, notification);
+    @override
```

Comment:
Should be `@Override` with uppercase `O` it seems

Reply:
oh no that's no good :open_mouth: that is why I want to get some of these merged :scream: I must have done something weird rebasing, I'll do a manual comparison to my rebased combined strange branch I actually used for build and make sure I didn't miss out anything important since apparently at least one rebase conflict ended up weird :grimacing:

Reply:
Well that probably mean we should introduce service compilation to our CI. But at the same time it's time/resources every time we add something new to it

Reply:
Wouldn't it be enough to add `service/main.py` to an existing test app? I think this particular bug should also have been triggered if there had been a service used in addition rather than a standalone service `service_only` test app

Reply:
Oh, fascinating! This is not actually a rebase error, it seems when using `service/main.py` (which is what I tested) this indeed isn't rendered out. So that's the reason, it needed a `service_only` test to show up

Reply:
Ah, it's probably the named extra services with the `--service` parameter. We have three ways to do services, I forgot (`service/main.py` with no name which is what I tested, `--service` with named services, and `service_only`)

Reply:
Yes indeed, I'm using the `--service service:service/main.py`, see from the the logs.
Complete command was:
```
/usr/bin/python3 -m pythonforandroid.toolchain apk --debug --bootstrap=sdl2 --dist_name etheroll --name Etheroll --version 2019.0624 --package com.github.andremiras.etheroll --android_api 27 --minsdk 21 --ndk-api 21 --private /home/andre/workspace/EtherollApp/.buildozer/android/app --permission INTERNET --permission WRITE_EXTERNAL_STORAGE --presplash /home/andre/workspace/EtherollApp/docs/images/etheroll-logo.png --icon /home/andre/workspace/EtherollApp/docs/images/icon.png --orientation portrait --window --service service:service/main.py --copy-libs --local-recipes /home/andre/workspace/EtherollApp/src/python-for-android/recipes --blacklist /home/andre/workspace/EtherollApp/blacklist.txt --arch armeabi-v7a --color=always --storage-dir="/home/andre/workspace/EtherollApp/.buildozer/android/platform/build" --ndk-api=21
```